tests/test_basic_ops_dict.py

```diff
@@ -733,3 +735,23 @@ def test_setdefault() -> None:
     assert cfg["foo"] == 10
 
     assert cfg["foo"] == 10
+
+
+def test_set_valid_union_value() -> None:
+    cfg = OmegaConf.structured(DictUnion)
+    cfg.dict = {"a": 4, "b": 3.14}
+    assert isinstance(cfg.dict._get_node("a"), UnionNode)  # type: ignore
```

Comment:
Could be removed since it is duplicated two lines below?

Reply:
Idem below.